src/metabase/models/cloud_migration.clj

```diff
@@ -229,7 +240,8 @@
   [{:keys [id external_id] :as migration} & {:keys [retry?]}]
   ;; dump-to-h2 starts behaving oddly if you try to dump repeatly to the same file
   ;; in the same process.
-  (let [dump-file (io/file (str "cloud_migration_dump_" (random-uuid) ".mv.db"))]
+  (let [dump-file (io/file (System/getProperty "java.io.tmpdir")
+                           (str "cloud_migration_dump_" (random-uuid) ".mv.db"))]
```

Comment:
How come we're dumping to a temp directory now? Maybe update the comment if there's a good reason

Reply:
Added:
```
  ;; The docker image process runs in non-root, so write to a dir it can access.
```

Was a rather nasty error mode actually.